platform/ios/os_ios.mm

```diff
@@ -273,13 +271,16 @@ void register_dynamic_symbol(char *name, void *address) {
 	return OK;
 }
 
-void OS_IOS::set_user_data_dir(String p_dir) {
-	Ref<DirAccess> da = DirAccess::open(p_dir);
-	user_data_dir = da->get_current_dir();
-	printf("setting data dir to %s from %s\n", user_data_dir.utf8().get_data(), p_dir.utf8().get_data());
-}
-
 String OS_IOS::get_user_data_dir() const {
+	static bool user_data_dir_set = false;
+	if (user_data_dir_set) {
```

Comment:
should this not be  `if( !user_data_dir_set )` ?  
as it is nothing can set this flag to true and you won't get the run-once behavior that appears to be intended.

Reply:
Oh, indeed. At least on M1 mac, it seems to do absolutely nothing and the resulting path is the same as the source, so I guess I have removed `!` when renaming the variable, and did not notice it. I'm not sure what's this code supposed to do anyway, so might worth removing it or use `NSSearchPathForDirectoriesInDomains` directly, like we do on macOS.